frame/oracle/src/lib.rs

```diff
@@ -692,11 +695,25 @@ pub mod pallet {
 				let min_accuracy = asset_info.threshold;
 				if accuracy < min_accuracy {
 					let slash_amount = asset_info.slash;
-					let try_slash = T::Currency::can_slash(&answer.who, slash_amount);
-					if !try_slash {
-						log::warn!("Failed to slash {:?}", answer.who);
+					let amount_staked = Self::oracle_stake(answer.who.clone())
+						.unwrap_or_else(|| 0_u32.into())
+						.saturating_sub(slash_amount);
+					OracleStake::<T>::insert(&answer.who, amount_staked);
```

Comment:
Do we need to insert if the `amount_staked` is zero?

Reply:
zero value is handled properly in function like remove_stake(), however it may be good to have None instead of zero, but I think both ways are fine and thus not doing any change for this in current PR.

Reply:
Can't we just remove the oracle when there is no stake?

Reply:
we can but don't want to do it in this PR. As I am not much confident about it the code.